src/components/track-reader-depth.tsx

```diff
@@ -0,0 +1,65 @@
+'use client';
+import {useEffect} from 'react';
+import {usePlausible} from 'next-plausible';
+
+import {debounce} from 'sentry-docs/utils';
+
+const EVENT = 'Read Progress';
+const milestones = [25, 50, 75, 100] as const;
+type Milestone = (typeof milestones)[number];
+type EVENT_PROPS = {page: string; readProgress: Milestone};
+
+export function ReaderDepthTracker() {
+  const plausible = usePlausible<{[EVENT]: EVENT_PROPS}>();
+
+  const sendProgressToPlausible = (progress: Milestone) => {
+    // TODO: remove this after PR review
+    const args = [
+      EVENT,
+      {props: {readProgress: progress, page: document.title}},
+    ] as const;
+    plausible(...args);
+    console.log('plausible event', ...args);
+  };
+
+  useEffect(() => {
+    const reachedMilestones = new Set<Milestone>();
+
+    const trackProgress = () => {
+      // calculate the progress based on the scroll position
+      const scrollPosition = window.scrollY;
+      const totalHeight = document.documentElement.scrollHeight - window.innerHeight;
+      let progress = Math.floor((scrollPosition / totalHeight) * 100);
+      // it's hard to trigger the 100% milestone, so we'll just assume beyond 95%
+      if (progress > 95) {
+        progress = 100;
+      }
+
+      // find the biggest milestone that has not been reached yet
+      const milestone = milestones.findLast(
+        m =>
+          progress >= m &&
+          !reachedMilestones.has(m) &&
+          // we shouldn't report smaller milestones once a bigger one has been reached
+          Array.from(reachedMilestones).every(r => m > r)
+      );
+      if (milestone) {
+        reachedMilestones.add(milestone);
+        sendProgressToPlausible(milestone);
+      }
+    };
+
+    // if the page is not scrollable, we don't need to track anything
+    if (document.documentElement.scrollHeight - window.innerHeight === 0) {
+      return () => {};
+    }
+    const debouncedTrackProgress = debounce(trackProgress, 20);
```

Comment:
As discussed offline: this prevents events from being sent when a user very quickly scrolls to the bottom and then leaves the page - which might not count as a read anyway

Reply:
yes, it doesn't happen if you scroll at any reasonable reading speed

Reply:
Have you considered using the IntersectionObserver API for this?

Reply:
I think the current approach is more precise

the marketing site / blog use the intersection observer and report 100% when the last heading is reached

a useful heuristic, but not very precise

Reply:
please keep in mind the measurement on the marketing sites is very rough (based on headers instead of the actual page scroll)

But I can do it @elijames-codecov

Reply:
turns out I had bug in my `debounce` logic, now it works as intended

it's simple, precise and performant

maybe we should port it to the marketing site @elijames-codecov once we make sure it works well?

Reply:
@a-hariti - Glad you got the debounce working! Since we're looking for insights as to which sections the users scroll into (to inform if there's seeing certain content and make content and some layout decisions), each section having a header, IntersectionObserver actually works well for our needs without added complexity and expensive operations...which is why we will continue to use it.